src/wix/WixToolset.Core/Compiler_Bundle.cs

```diff
@@ -2422,6 +2444,18 @@ private string ParseChainPackage(XElement node, WixBundlePackageType packageType
 
                 switch (packageType)
                 {
+                    case WixBundlePackageType.Bundle:
+                        WixBundleBundlePackageAttributes bundleAttributes = 0;
+
+                        this.Core.AddSymbol(new WixBundleBundlePackageSymbol(sourceLineNumbers, id)
+                        {
+                            Attributes = bundleAttributes,
+                            InstallCommand = installArguments,
+                            RepairCommand = repairArguments,
+                            UninstallCommand = uninstallArguments,
```

Comment:
Should we default the `UninstallCommand` for Bundles if they are not permanent?

Reply:
The Burn engine automatically adds `/uninstall`. `BundlePackage`s are always uninstallable and always repairable.

Reply:
Cool. Two questions:
1. Should `UninstallCommand` be additional arguments to `/uninstall` in the engine or if specified should it take over fully?
2. I can see scenarios where devs will want to not repair the nested bundle. Today with ExePackage that is done by not adding the `RepairArguments`. Is this supported for BundlePackage without custom BA logic?

Reply:
`BundlePackage` is a lot closer to `MsiPackage` than `ExePackage` for these things.
1. Additional. Similar to `MsiProperty` for `MsiPackage` - the engine is going to add certain ones that the bundle author and BA have no control over.
2. There's the same problem with `MsiPackage`. It sounded like we were punting that to v5.

Reply:
1. Cool. I wonder if we should name the attribute differently so developers don't set  `UninstallArguments="-uninstall"`  regularly.
2. True but redists are more likely to be shipped as Exes and Bundles than MSI packages. For example, `vc_redist` is a Bundle. The automatic detection of the Bundle will be awesome but most devs would not want to repair `vc_redist` with their bundle. I think `BundlePackage` should be considered more like `ExePackage` than `MsiPackage`.

Reply:
Sounds like we need to design and implement `RepairCondition` into v4 then.

Reply:
> Sounds like we need to design and implement RepairCondition into v4 then.

We could do that. But would the same repair (non)behavior as `ExePackage` work in this case? No `RepairArguments` means no repair.

Reply:
No. The Burn engine knows how to do a repair, just like it knows how to repair an MsiPackage. We're not going to do weird things to avoid `RepairCondition`.

Reply:
Okay.

However, I do think it is better to think of `BundlePackage` more like `ExePackage` than `MsiPackage`. The use cases for `BundlePackage` are much more aligned with `ExePackage`.

Reply:
The use cases might be more aligned with `ExePackage`, but the capabilities of the package type itself is much more closely aligned with `MsiPackage`.

Reply:
I understand that and I can see why you are approaching things the way you are given the Burn implementation. I'm approaching the developer experience side (use cases) and thus focused on how the language exposes all of the awesome Burn functionality.

Reply:
I am also focused on making sure the language exposes all of the awesome Burn functionality, which is why I think it is a mistake to look towards `ExePackage` for inspiration when it is not a good example of how it can be used.

Reply:
Okay, cool. 

I believe the most important use case to keep in mind is that most bundles contained in other bundles are redists. We've learned that repairing redists in the bundle is a bad idea. Repairing redists should be avoided enough that we might even consider making no-repair the default for bundles (though I'm not necessarily sold on that idea myself yet).

Less important but something I noodle on: bundles are always visible in ARP whereas MSI packages are invisible by default. That changes the interaction with uninstall since the user can manually execute it on nested bundles. At this point, I do think that the `Permanent` attribute plus dependency tracking exposes the right control for uninstall of nested bundles. Of course, if we hid nested bundles by default lots would need to be reconsidered.

Reply:
For BundlePackage ARP visibility, that is a feature tracked in https://github.com/wixtoolset/issues/issues/4454. I was going to implement that after merging, so if you have thoughts on that please put them there.

Reply:
Interesting. I was not expecting 4454 to be moved into `v4.0`.

Hiding nested bundles changes quite a few things. If hiding nested bundles was the default then `BundlePackage` _would_ be very much like `MsiPackage`. But I'm not sure nested bundles should be hidden by default. We'll definitely need to think through all the ramifications of hidden bundles. For example, I'm not sure nested bundles should be hidden by default.

Reply:
Clearly... I'm not sure nested bundles should be hidden by default. 🤣